### ISSUE TITLE
Type-inference problems with length_count!(value!(..), ..)

### ISSUE THREAD

**Issue by lukaslueg** [Geal/nom#771]
- Rust version : rustc 1.26.0 (a77568041 2018-05-07)
- nom version : 4.0.0

Silly example that demonstrates using `value!()` as the first argument of `length_count()`:

```
#[macro_use]
extern crate nom;

named_args!(parse_lcv(count: usize)<Vec<u32>>,
    length_count!(value!(count), call!(nom::be_u32))
);

```

The above fails to compile, apparently because `value!()` does not determine the (unused) error-type:

> error[E0283]: type annotations required: cannot resolve `u32: std::convert::From<_>`
> ...
> 121 | /         named_args!(parse_lcv(count: usize)<Vec<u32>>,
> 122 | |             length_count!(value!(count), call!(nom::be_u32))
> 123 | |         );
>     | |__________^
>     |
>     = note: required because of the requirements on the impl of `nom::Convert<nom::Err<&[u8], _>>` for `nom::Err<&[u8]>`
>     = note: required by `nom::Convert::convert`

My solution is to wrap `value!()` as `add_return_error!(ErrorKind::Custom(0u32), value!(...))`, which is essentially a no-op because `value!()` with a single value can't fail anyway. The `0u32` determines the error-type to `u32`, which makes the whole thing compile, though.

**Comment by Geal** [comment]
hello, this is fixed in ed1abaf

**Comment by lukaslueg** [comment]
thanks
